src/main/java/ca/uhn/fhir/jpa/starter/MyPlainProvider.java

```diff
@@ -0,0 +1,157 @@
+import java.util.List;
+
+public class MyPlainProvider {
+  @Operation(name="$process-message")
+  public IBaseResource processMessage(
+      @OptionalParam(name="async") String async,
+      @OptionalParam(name="response-url") String responseUrl,
+      @ResourceParam Bundle bundleR4
+  ) {
+    // validate the bundle
+    if (bundleR4.getType() != Bundle.BundleType.MESSAGE) {
+      OperationOutcome oo = new OperationOutcome();
+      oo.addIssue()
+          .setSeverity(OperationOutcome.IssueSeverity.ERROR)
+          .setDiagnostics("Bundle is not of type 'message'");
+      return oo;
+    }
+
+    if (!bundleR4.hasEntry()) {
+      OperationOutcome oo = new OperationOutcome();
+      oo.addIssue()
+          .setSeverity(OperationOutcome.IssueSeverity.ERROR)
+          .setDiagnostics("Bundle has no entries, a MessageHeader is required");
+      return oo;
+    }
+
+    // check that there is a message header
+    Resource firstResource = bundleR4.getEntryFirstRep().getResource();
+    if ( firstResource.getResourceType() != ResourceType.MessageHeader) {
+      OperationOutcome oo = new OperationOutcome();
+      oo.addIssue()
+          .setSeverity(OperationOutcome.IssueSeverity.ERROR)
+          .setDiagnostics("First resource in message bundle must be of type 'MessageHeader'");
+      return oo;
+    }
+
+    MessageHeader msgHead = (MessageHeader) firstResource;
+
+    // check messageHeader is well formed
+    if (!msgHead.hasSource()) {
+      OperationOutcome oo = new OperationOutcome();
+      oo.addIssue()
+          .setSeverity(OperationOutcome.IssueSeverity.ERROR)
+          .setDiagnostics("'MessageHeader' is missing source");
+      return oo;
+    } else if (!msgHead.getSource().hasEndpoint()){
+      OperationOutcome oo = new OperationOutcome();
+      oo.addIssue()
+          .setSeverity(OperationOutcome.IssueSeverity.ERROR)
+          .setDiagnostics("'MessageHeader.source' is missing an endpoint");
+      return oo;
+    }
+
+
+    System.out.println(async);
+    if (async.equals("true")) {
+      // asynchronous
+      String asyncResponseUrl;
+      if (responseUrl != null) {
+        asyncResponseUrl = responseUrl;
+      } else {
+        asyncResponseUrl = msgHead.getSource().getEndpoint();
+      }
+      // do any async operations in the new thread
+      Thread newThread = new Thread(() -> {
+        MessageHeader.ResponseType responseType = doEvent(msgHead);
+        Bundle bundle = buildResponseMessage(msgHead, responseType);
+        FhirContext ctx = FhirContext.forR4();
+        IParser parser = ctx.newJsonParser();
+        String bundleString = parser.encodeResourceToString(bundle);
+        makePost(asyncResponseUrl, bundleString);
+      });
+      newThread.start();
+      OperationOutcome oo = new OperationOutcome();
+      oo.addIssue()
+          .setSeverity(OperationOutcome.IssueSeverity.INFORMATION)
+          .setDiagnostics("Asynchronous message successfully received");
+      return oo;
```

Comment:
The [$process-message docs](https://www.hl7.org/fhir/messageheader-operation-process-message.html) are a little confusing but I don't think this should return an OperationOutcome. 
> The following rules apply when using the $process-message operation asynchronously:
-The URL has at least one parameter: http://server/base/$process-message?async=true
-The server acknowledges the message with a 200 OK with no body, or returns an HTTP error if the message cannot be processed
-Accepting the message means that the server has understood the message enough to know where to respond
-An OperationOutcome SHOULD be returned in either case

The second bullet point says it should response with 200 OK and no body. I'm personally confused with the last bullet point because there is no clear "either case". Not sure what that is referring too... but if you scroll further down to the [examples](https://www.hl7.org/fhir/messageheader-operation-process-message.html#examples) it shows the asynchronous response of 200 OK with no body. So I _think_ that is the correct response here.

Reply:
That part is confusing to me too, since it says it should respond with an operation outcome, unfortunately I'm not sure how to return an empty body from HAPI's handler, since it needs to return an `IFhirResource`

Reply:
That's a good point so I did some digging. I tried a couple things but couldn't get it to work. I haven't tried implementing it yet but section 4.6.7 on this page looks like it might be the way to go https://hapifhir.io/hapi-fhir/docs/server_plain/rest_operations_operations.html
I think it would be good to get it done in this PR but if it doesn't work then we can just throw a task in the backlog. At least for our clients it shouldn't break anything by including a body when we don't expect one